grammar.js

```diff
@@ -674,13 +674,20 @@ module.exports = grammar({
         $.super,
         alias($._reserved_identifier, $.identifier)
       )),
-      '.',
+      choice('.', '?.'),
       field('property', alias($.identifier, $.property_identifier))
     )),
 
     subscript_expression: $ => prec.right(PREC.MEMBER, seq(
-      field('object', choice($._expression, $.super)),
-      '[', field('index', $._expressions), ']'
+      field('object', choice(
+        $._expression,
+        $.identifier,
+        $.super,
+        alias($._reserved_identifier, $.identifier)
+      )),
+      choice('[', seq('?.', '[')),
```

Comment:
I think it would be slightly clearer to write this is as:

```js
optional('?.'),
'[',
```

instead of

```js
choice('[', seq('?.', '[')),
```